python/cudaq/kernel/ast_bridge.py

```diff
@@ -3377,73 +3377,96 @@ def visit_Compare(self, node):
         comparator = self.popValue()
```

Comment:
`comparator` should be renamed `right` to improve readability of the rest of this.